cmd/thanos/bucket.go

```diff
@@ -213,4 +217,118 @@ func registerBucket(m map[string]setupFunc, app *kingpin.Application, name strin
+		Default("").String()
+	sortBy := inspect.Flag("sort-by", "Sort by columns.").
+		Default("FROM,UNTIL").String()
+
+	m[name+" inspect"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer, _ bool) error {
+		bucketConfig, err := objStoreConfig.Content()
+		if err != nil {
+			return err
+		}
+
+		bkt, err := client.NewBucket(logger, bucketConfig, reg, name)
+		if err != nil {
+			return err
+		}
+
+		// Dummy actor to immediately kill the group after the run function returns.
+		g.Add(func() error { return nil }, func(error) {})
+
+		defer runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+
+		// Getting Metas
+		var blockMetas []*block.Meta
+		err = bkt.Iter(ctx, "", func(name string) error {
+			id, ok := block.IsBlockDir(name)
+			if !ok {
+				return nil
+			}
+
+			m, err := block.DownloadMeta(ctx, logger, bkt, id)
+			if err != nil {
+				return err
+			}
+
+			blockMetas = append(blockMetas, &m)
+
+			return nil
+		})
+
+		if err != nil {
+			return err
+		}
+
+		return printTable(blockMetas, replica, sortBy)
+	}
+}
+
+func printTable(blockMetas []*block.Meta, replica *string, sortBy *string) error {
+	header := []string{"ULID", "FROM", "UNTIL", "RANGE", "UNTIL-COMP", "~SIZE", "#SERIES", "#SAMPLES", "#CHUNKS", "COMP-LEVEL", "COMP-FAILED", "REPLICA", "RESOLUTION", "SOURCE"}
+
+	var lines [][]string
+	p := message.NewPrinter(language.English)
+
+	for _, blockMeta := range blockMetas {
+		if *replica != "" && *replica != blockMeta.Thanos.Labels["replica"] {
+			continue
+		}
+
+		var line []string
+		line = append(line, blockMeta.ULID.String())
+		line = append(line, time.Unix(blockMeta.MinTime/1000, 0).Format("02-01-2006 15:04:05"))
+		line = append(line, time.Unix(blockMeta.MaxTime/1000, 0).Format("02-01-2006 15:04:05"))
+		timeRange := time.Duration((blockMeta.MaxTime - blockMeta.MinTime) * 1000 * 1000)
+		line = append(line, timeRange.String())
+		untilComp := "-"
+		if blockMeta.Thanos.Downsample.Resolution == 0 { // data currently raw, downsample if range >= 40 hours
+			untilComp = (time.Duration(40*60*60*1000*time.Millisecond) - timeRange).String()
+		}
+		if blockMeta.Thanos.Downsample.Resolution == 5*60*1000 { // data currently 5m resolution, downsample if range >= 10 days
+			untilComp = (time.Duration(10*24*60*60*1000*time.Millisecond) - timeRange).String()
+		}
+		line = append(line, untilComp)
+		line = append(line, p.Sprintf("%0.2fMiB", (float64(blockMeta.Stats.NumSamples)*1.07)/(1024*1024)))
+		line = append(line, p.Sprintf("%d", blockMeta.Stats.NumSeries))
+		line = append(line, p.Sprintf("%d", blockMeta.Stats.NumSamples))
+		line = append(line, p.Sprintf("%d", blockMeta.Stats.NumChunks))
+		line = append(line, p.Sprintf("%d", blockMeta.Compaction.Level))
+		line = append(line, p.Sprintf("%t", blockMeta.Compaction.Failed))
+		line = append(line, blockMeta.Thanos.Labels["replica"])
```

Comment:
same as 3rd comment above

Reply:
As above: I added a labels column which can be filtered via --selector or -l (similar to kubectl get po -l key=value)